File: Projet_final/projetFinalSpringBoot/src/main/java/com/exemple/model/Module.java
package com.exemple.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name = "Module")
public class Module {
	@Id
	@SequenceGenerator(name = "seqMateriel", sequenceName = "seq_materiel", initialValue = 101, allocationSize = 1)
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqMateriel")
	@Column(name = "Module_id")
	@JsonView(JsonViews.Common.class)
	private Long id;
	@Column(name = "Module_dates")
	@JsonView(JsonViews.Common.class)
	private Date[] dates;
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "Module_matiere")
	@JsonView(JsonViews.Module.class)
	private Matiere matiere;
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "Module_formateur")
	@JsonView(JsonViews.Module.class)
	private Formateur formateur;
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "Module_cursus")
	@JsonView(JsonViews.Module.class)
	private Cursus cursus;
	@Version
	private int version;

	public Matiere getMatiere() {
		return matiere;
	}

	public void setMatiere(Matiere matiere) {
		this.matiere = matiere;
	}

	public Formateur getFormateur() {
		return formateur;
	}

	public void setFormateur(Formateur formateur) {
		this.formateur = formateur;
	}

	public Date[] getDates() {
		return dates;
	}

	public void setDates(Date[] dates) {
		this.dates = dates;
	}

	public Cursus getCursus() {
		return cursus;
	}

	public void setCursus(Cursus cursus) {
		this.cursus = cursus;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}
}

File: Projet_final/angularBT/src/app/model/module.ts
import { Cursus } from './cursus';
import { Formateur } from './formateur';
import { Matiere } from './matiere';
export class Module {
  
   constructor(private _id:string='',
  private _dates: Date=null,
  private _matiere:Matiere=null,
  private _formateur:Formateur=null,
  private _cursus:Cursus=null){
    
  }
  
  public get id(): string {
    return this._id;
  }
  
  public set id(_id: string) {
    this._id =_id;
  }
  
  public get dates(): Date {
    return this._dates;
  }
  
  public set dates(_dates: Date) {
    this._dates =_dates;
  }
  
  public get matiere(): Matiere {
    return this._matiere;
  }
  
  public set matiere(_matiere: Matiere) {
    this._matiere =_matiere;
  }
  
  public get formateur(): Formateur {
    return this._formateur;
  }
  
  public set formateur(_formateur: Formateur) {
    this._formateur =_formateur;
  }
  
  public get cursus(): Cursus {
    return this._cursus;
  }
  
  public set cursus(_cursus: Cursus) {
    this._cursus =_cursus;
  }
}